### DISCUSSION TITLE
Convert the web app into a Progressive Web App

### DISCUSSION THREAD

**Discussion by albertsalles4** [ghostfolio/ghostfolio#1412] [Ideas]
I think it would be great to convert the website into a PWA so that it can be installed natively as a Desktop or Mobile application. It should be an easy feature to implement.

**Comment by dtslvr** [comment]
Hi @albertsalles4

Thank you for coming up with this idea. Would you like to work on the implementation?

By the way, there is already a [site.webmanifest](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/assets/site.webmanifest) in the project.

**Reply by albertsalles4** [reply]
Sure, will do. 👍

**Reply by dtslvr** [reply]
Great! Here is some more background on this: I have already setup `@angular/pwa` for Ghostfolio once, but I ran into a problem after every deployment: https://stackoverflow.com/q/65277756/744012

That is why I have removed it again. I don't know if this is still an issue, though.

**Comment by dtslvr** [comment]
Implemented via #1436.

**Comment by secondl1ght** [comment]
The app isn't working for me as a PWA, was this feature removed again?

**Reply by dtslvr** [reply]
No, the feature is still there.

<img width="673" alt="lighthouse-pwa-report" src="https://github.com/ghostfolio/ghostfolio/assets/4159106/a1c213e4-2444-4312-8ece-e19146b6dddf">

What is not working?

**Reply by secondl1ght** [reply]
It looks like there are some errors in the console and one of them is the manifest not loading:

![image](https://github.com/ghostfolio/ghostfolio/assets/85003930/4786e07f-7b7c-4e98-b7d7-11c9099b3117)

**Reply by dtslvr** [reply]
Seems to be an issue related to _Umbrel_: `http://umbrel.local:2000` vs. `http://umbrel.local:3334`

**Reply by secondl1ght** [reply]
Should I open a bug report in this repo or in Umbrel?

**Reply by dtslvr** [reply]
As it works in the official deployment, I would report it there.

**Reply by secondl1ght** [reply]
Ok I think it would still be up to the Ghostfolio maintainers to make the app work for Umbrel, I don't think the Umbrel devs do any work on making apps compatible with their platform. But I will put a report in and see what they say.

**Reply by secondl1ght** [reply]
https://github.com/getumbrel/umbrel-apps/issues/934

**Comment by MyWay** [comment]
Yeah apparently before it was working, but now not anymore.

**Comment by bido1990** [comment]
Hi, is there any update on the PWA status? I tried mounting a manifest.json in my docker compose, but it's not been loaded, and i don't know how to proceed further

**Comment by bido1990** [comment]
I managed to get It working again. The issue was that querying for the manifest.json at https://your.domain/manifest.json was giving me an empty response. The reason is because The manifest.json is served athttps://your.domain/api/assets/en/site.webmanifest
At that address using curl I got the results expected, and on Android I can install PWA. If using traefik+authentik you need to bypass auth for the specific path
